libcxx/include/__iterator/prev.h

```diff
@@ -35,6 +39,13 @@ prev(_InputIter __x, typename iterator_traits<_InputIter>::difference_type __n =
   return __x;
 }
 
+template <class _BidirectionalIterator,
+          __enable_if_t<__has_bidirectional_iterator_category<_BidirectionalIterator>::value, int> = 0>
+[[__nodiscard__]] _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX17 _BidirectionalIterator
+prev(_BidirectionalIterator __it) {
+  return std::prev(std::move(__it), 1);
+}
```

Comment:
Can you mention [LWG3197](https://cplusplus.github.io/LWG/issue3197) and explain that while it's not 100% clear yet what the Standard wants us to do, we take the stance that a non-bidi iterator is ill-formed because it's likely to lead to a bug.

Reply:
[LWG4100](https://cplusplus.github.io/LWG/issue4100) might also be related. It's unclear whether additional signatures of standard free functions are allowed. And it's squirrelly that two such overloads have different conditions on well-formedness.